python/ray/tune/tune.py

```diff
@@ -305,12 +305,7 @@ def run(
     storage_path: Optional[str] = None,
     search_alg: Optional[Union[Searcher, SearchAlgorithm, str]] = None,
     scheduler: Optional[Union[TrialScheduler, str]] = None,
-    keep_checkpoints_num: Optional[int] = None,
-    checkpoint_score_attr: Optional[str] = None,
-    checkpoint_freq: int = 0,
-    checkpoint_at_end: bool = False,
-    checkpoint_keep_all_ranks: bool = False,
-    checkpoint_upload_from_workers: bool = False,
+    checkpoint_config: Optional[CheckpointConfig] = None,
```

Comment:
We may need to keep all those removed args for 2 releases. Since tune.run is a "stable" public API.

Reply:
yeah, for sure.
I added them back, and added a bunch of deprecation warnings.